morphia/src/main/java/org/mongodb/morphia/query/CountOptions.java

```diff
@@ -0,0 +1,261 @@
+    }
+
+    /**
+     * Sets the maximum execution time on the server for this operation.
+     *
+     * @param maxTime  the max time
+     * @param timeUnit the time unit, which may not be null
+     * @return this
+     */
+    public CountOptions maxTime(final long maxTime, final TimeUnit timeUnit) {
+        notNull("timeUnit", timeUnit);
+        options.maxTime(maxTime, timeUnit);
+        return this;
+    }
+
+    /**
+     * Returns the readPreference
+     *
+     * @return the readPreference
+     */
+    public ReadPreference getReadPreference() {
+        return options.getReadPreference();
+    }
+
+    /**
+     * Sets the readPreference
+     *
+     * @param readPreference the readPreference
+     * @return this
+     */
+    public CountOptions readPreference(final ReadPreference readPreference) {
+        options.readPreference(readPreference);
+        return this;
+    }
+
+    /**
+     * Returns the readConcern
+     *
+     * @return the readConcern
+     * @mongodb.server.release 3.2
+     */
+    public ReadConcern getReadConcern() {
+        return options.getReadConcern();
+    }
+
+    /**
+     * Sets the readConcern
+     *
+     * @param readConcern the readConcern
+     * @return this
+     * @mongodb.server.release 3.2
+     */
+    public CountOptions readConcern(final ReadConcern readConcern) {
+        options.readConcern(readConcern);
+        return this;
+    }
+
+    /**
+     * Returns the collation options
+     *
+     * @return the collation options
+     * @mongodb.server.release 3.4
+     */
+    public Collation getCollation() {
+        return options.getCollation();
+    }
+
+    /**
+     * Sets the collation
+     *
+     * @param collation the collation
+     * @return this
+     * @mongodb.server.release 3.4
+     */
+    public CountOptions collation(final Collation collation) {
+        options.collation(collation);
+        return this;
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (!(o instanceof FindOptions)) {
+            return false;
+        }
+
+        final CountOptions that = (CountOptions) o;
+
+        return getLimit() == that.getLimit()
+            && getMaxTime(TimeUnit.MILLISECONDS) == that.getMaxTime(TimeUnit.MILLISECONDS)
+            && getSkip() == that.getSkip()
+            && (getHint() == null && that.getHint() == null || getHint().equals(that.getHint()))
+            && (getReadPreference() == null && that.getReadPreference() == null || getReadPreference().equals(that.getReadPreference()))
+            && (getReadConcern() == null && that.getReadConcern() == null || getReadConcern().equals(that.getReadConcern()))
+            && (getCollation() == null && that.getCollation() == null || getCollation().equals(that.getCollation()));
+
+    }
+
+    @Override
+    public int hashCode() {
+        return getOptions().hashCode();
```

Comment:
Even though equals is for testing, if you're going to have equals and hashCode in a public class, they shouldn't violate the equals-consistent-with-hash-code rule.  So this still needs to be addressed.